allennlp/models/coreference_resolution/coref.py

```diff
@@ -602,8 +603,19 @@ def decode(self, output_dict: Dict[str, torch.Tensor]):
             which are in turn comprised of a list of (start, end) inclusive spans into the
             original document.
         """
+
+        # A tensor of shape (batch_size, num_spans_to_keep, 2), representing
+        # the start and end indices of each span.
         batch_top_spans = output_dict["top_spans"].data.cpu()
+
+        # A tensor of shape (batch_size, num_spans_to_keep) representing, for each span,
+        # the index into ``antecedent_indices`` which specifies the antecedent span. Additionally,
+        # the index can be -1, specifying that the span has no predicted antecedent.
         batch_predicted_antecedents = output_dict["predicted_antecedents"].data.cpu()
+
+        # A tensor of shape (num_spans_to_keep, max_antecedents), representing the indices
```

Comment:
Possible modification: A tensor of shape `(num_spans_to_keep, max_antecedents)`, representing _which_ antecedents were considered for each span.  Each element is an index between 0 and `num_spans_to_keep`, so if `antecedent_indices[i, j] = k` for some `j`, it means that span `i` considered span `k` as a possible antecedent.